### ISSUE TITLE
How can Solo work when using in external class

### ISSUE THREAD

**Issue by renas** [RobotiumTech/robotium#124]
_From [entrang2...@gmail.com](https://code.google.com/u/112725656854968657886/) on December 14, 2010 01:59:01_

Create a common class such as "common.java" with using solo to do something.

In the test case class there is a test case with call a method which using solo from the "common.java" class. 

However, it does not work when running this test because it's does not recognize the solo.

Please help me to solve this problem.
It's appreciate if there is any example.

Thank so much.
Hakuen

_Original issue: http://code.google.com/p/robotium/issues/detail?id=45_

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on December 14, 2010 02:02:18_

**Owner:** renasreda  
  **Labels:** -Type-Defect Type-Other

**Comment by renas** [comment]
_From [anandpai...@gmail.com](https://code.google.com/u/116639303515044083042/) on December 14, 2010 03:41:37_

u should pass the solo object or u have to create solo object in your Common.java

Example:

tescase.java has your test method

msolo = Solo(getInstrumentation(), getActivity());

test1() {
    Common.clickontext(solo,text);
}

Common.java should contain.

public static void clickontext(Solo solo,String text) {
  solo.clickOnText(text);
}

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on December 16, 2010 07:53:59_

Please see above answer.

**Status:** Done

**Comment by renas** [comment]
_From [kpsrika...@gmail.com](https://code.google.com/u/101065684543152900811/) on November 24, 2011 02:17:20_

I did write my tests as suggested by Anand. In the above scenario let's say I have created a bunch of methods in Common.java and also I created multiple classes similar to Common.java which provide different reusable functionality.
Each time I invoke a method in Common.java or other classes I am passing solo object(from testcase.java), instead is there a way to create a solo object in Common.java.
Anand did mention that this can be done but an example/suggestion would be great.
